torch/_inductor/kernel/flex_attention.py

```diff
@@ -121,51 +66,43 @@ def build_subgraph_buffer(
         # expect that these are lifted inputs that fill up the '*other_buffers'
         # tuple and already have corresponding TensorBoxes passed in as args.
         if node.op == "placeholder":
-            is_lifted_input = cnt >= len(placeholder_inps)
-            lifted_input_index = index_to_other_buffers(cnt, graph_type)
-            env[node] = (
-                args[lifted_input_index] if is_lifted_input else placeholder_inps[cnt]
-            )
+            env[node] = args[cnt]
             cnt += 1
         elif node.op == "call_function":
             # For call_function we use the default lowerings and pass in the
             # already created TensorBoxes as args
-            from torch.utils._pytree import tree_map
 
             args, kwargs = tree_map(
                 lambda x: env[x] if x in env else x, (node.args, node.kwargs)
             )
             env[node] = lowerings[node.target](*args, **kwargs)
         elif node.op == "output":
-            # For the output node we need to create a ComputedBuffer
-            # which represents the actual score modification
-            # The joint_graph's output should be of the form[grad_score, None, None, None, None]
-            # This is because only the 'score' requires grad and the other outputs are
-            # the non-differentiable index scalars
-            if graph_type == SubgraphType.FWD or graph_type == SubgraphType.JOINT_FWD:
-                output_node = node.args[0]
-            else:
-                output_node = node.args[0][0]
-            output_buffer = env[output_node]
-            assert isinstance(output_buffer, TensorBox), (
-                "The output node  for flex attention's subgraph must be a TensorBox, but got: ",
-                type(output_buffer),
-            )
-            assert isinstance(output_buffer.data, StorageBox), (
-                "The output node for the flex attention subgraph must be a StorageBox, but got: ",
-                type(output_buffer),
-            )
-            # Create the ComputedBuffer directly that will be inlined into the modification block
-            subgraph_buffer = ComputedBuffer(
-                name=None,
-                layout=FlexibleLayout(
-                    device=output_buffer.data.get_device(),
-                    dtype=output_buffer.data.get_dtype(),
-                    size=output_buffer.data.get_size(),
-                ),
-                data=output_buffer.data.data,  # type: ignore[arg-type]
-            )
-            return subgraph_buffer
+
+            def convert_output_node_to_buffer(output):
+                if output is None:
+                    return None
+                output_node = output
+                output_buffer = env[output_node]
```

Comment:
also why dont we need the node[0][0] is has the output been flattened? e.g. ahhh as I am writing this I realized why you tree_map

could we also bake this into some comment